Feature: Add (optional) wind forecast

pauln opened the pull request:
Adds an option to add wind speed and direction to the weather bar.  It's opt-in so that existing users won't see any unexpected change.

decompil3d commented on src/hourly-weather.ts:
```diff
@@ -321,6 +326,30 @@ export class HourlyWeatherCard extends LitElement {
+  }
+
+  private formatWindDir(degrees: number): string {
+    const directions = ['N','NNE','NE','ENE','E','ESE','SE','SSE','S','SSW','SW','WSW','W','WNW','NW','NNW','N'];
```
Can you make these localizable? Other languages may use different cardinal direction names.

pauln replied:
Is there a clean way to make an array of strings localizable, or am I best to add a separate language string for each direction and compose those into an array to use them?

decompil3d replied:
Probably best to make a separate object of loc strings in the locale jsons (e.g. direction.n) and then compose into an array at runtime -- ideally memoized into a member variable so you only have to compose it once as long as language doesn't change. I do that with the condition strings too I think, if you want to follow that pattern.

pauln replied:
I've used the condition strings as a guide, but deviated a little so that the member variable is the array (so there doesn't need to be an intermediate variable / another step to get to the array).  Does that look reasonable to you?

decompil3d commented on src/weather-bar.ts:
```diff
@@ -26,6 +29,9 @@ export class WeatherBar extends LitElement {
   hide_temperatures = false;
 
+  @property({ type: Boolean })
+  show_wind = true;
```
```suggestion
  show_wind = false;
```

decompil3d reviewed (changes requested):
Thanks for this great contribution. Just a couple of comments.

decompil3d commented:
Oh, and please also add tests for this new functionality.

pauln commented:
Ah, sorry - I didn't see the tests.  I'll have a go at adding some tests later (or perhaps tomorrow).

decompil3d commented on src/hourly-weather.ts:
```diff
@@ -82,6 +84,9 @@ export class HourlyWeatherCard extends LitElement {
   private labelsLocalized = false;
 
+  private _directions = Object.keys(DIRECTIONS);
+  private directionsLocalized = false;
```
Make sure to set to `false` after line 96 just like we do for the `labelsLocalized` variable.

decompil3d reviewed (changes requested):
Thanks -- small comment for the directions and then just waiting on tests.

pauln commented:
I've implemented the fix, and added tests for not showing wind by default and showing wind (based on existing test data, though I've added the wind speed unit) if enabled.  Are there any other tests you'd particularly like to see for this PR?

decompil3d commented:
Going to merge this and then do a fast-follow to add other locale translations and maybe adjust styling a smidge. Thanks again for your contribution. I'll try to comment here with the version to expect the change in.

decompil3d commented:
Released in 4.5.0 -- should be up in HACS shortly.
